### PR TITLE
Landing Page IN and Pagination

### PR DESCRIPTION
**Pull request by rajivtitus**
### What this PR does (required):
- Search profiles from Landing Page - redirects to dashboard
- Users can now search for profiles when they are not logged in.
- Users can 'search by city' or between a 'specific date range'
- Implemented Pagination in Dashboard 
- Only logged in users can send requests

### Screenshots / Videos (required):
- 

- 

- 

### Any information needed to test this feature (required):
- Enter city/date range in landing page and view the list of profiles
- If you want to check if pagination works, we need to create 10-15 new profiles or change `itemsPerPage` to 1 in `ProfileList` component

### Any issues with the current functionality (optional):
- None

### CODE REVIEW

File: client/src/pages/LandingPage/LandingPage.tsx [hunk 1]
```diff
@@ -33,6 +63,7 @@ export default function LandingPage(): JSX.Element {
                   format="MM/dd/yyyy"
                   value={startDate}
                   onChange={(date) => setStartDate(date)}
+                  required={true}
```
Comment (xia0m) [inline client/src/pages/LandingPage/LandingPage.tsx]: can simply use `required` here

File: client/src/pages/LandingPage/LandingPage.tsx [hunk 2]
```diff
@@ -45,10 +76,11 @@ export default function LandingPage(): JSX.Element {
                   format="MM/dd/yyyy"
                   value={endDate}
                   onChange={(date) => setEndDate(date)}
+                  required={true}
```
Comment (xia0m) [inline client/src/pages/LandingPage/LandingPage.tsx]: same

File: client/src/pages/LandingPage/LandingPage.tsx [hunk 3]
```diff
@@ -18,9 +40,17 @@ export default function LandingPage(): JSX.Element {
...
             <Typography className={classes.label}>Where</Typography>
-            <TextField variant="outlined" placeholder="Anywhere" className={classes.input} />
+            <TextField
+              onChange={(e) => setCity(e.target.value)}
+              value={city}
+              variant="outlined"
+              color="secondary"
+              placeholder="Anywhere"
+              className={classes.input}
+              required={true}
```
Comment (xia0m) [inline client/src/pages/LandingPage/LandingPage.tsx]: can simply use required here

**Review by xia0m — APPROVED** [review]

**Comment by rajivtitus** [comment]
### Added mobile responsiveness for Landing Page.
Capture

Capture

File: client/src/pages/LandingPage/LandingPage.tsx (hunk 1 above)
Comment (rajivtitus) [reply client/src/pages/LandingPage/LandingPage.tsx]: Edited accordingly, thank you!

File: client/src/pages/LandingPage/LandingPage.tsx (hunk 3 above)
Comment (rajivtitus) [reply client/src/pages/LandingPage/LandingPage.tsx]: changed all throughout

**Pull request merged**